src/Workspaces/Core/Portable/Workspace/Solution/TextDocument.cs

```diff
@@ -17,11 +20,7 @@ public class TextDocument
         /// <summary>
         /// The project this document belongs to.
         /// </summary>
-        public Project Project { get; protected set; }
-
-        protected TextDocument()
```

Comment:
This is technically a breaking change, but since this constructor never assigned the State property, deriving from the class would have been useless.